### PR TITLE
Allow keepNames esbuild config option

### PR DESCRIPTION
**Pull request by KinnersleyM**
This PR allows for the [`keepNames`](https://esbuild.github.io/api/#keep-names) esbuild option.

It also allows for some extensibility for future esbuild options this project might want to support.

Error message for using an invalid option:
image

Without `keepNames`
```js
var RandomType = class RandomType2 {
};
```
With `keepNames`
```js
var RandomType = /* @__PURE__ */ __name(class RandomType2 {
}, "RandomType");
```

This fixes an issue I have been having with using the `@anatine/esbuild-decorators` plugin to use `"emitDecoratorMetadata": true` in my tsconfig. Prior to this, my `type-graphql` types were coming back as:
image
but with keepNames:
image

### CODE REVIEW

**Comment by thdxr** [comment]
@KinnersleyM thanks for the PR - it looks good! There's a conflict that needs to be resolved but I'll merge it after then

**Comment by KinnersleyM** [comment]
@thdxr Thank you! Just rebased now 👍

File: packages/cli/scripts/util/cdkHelpers.js [hunk 1]
```diff
@@ -193,13 +193,17 @@ async function loadEsbuildConfigOverrides(configPath) {
   }
   const configOverrides = require(configFullPath);
 
-  // validate only "plugins" can be overrid
+  const allowedOptions = ["plugins", "keepNames"];
+
+  // validate only allowed options can be overridden
   const nonPluginsKey = Object.keys(configOverrides).find(
```
Comment (fwang) [inline packages/cli/scripts/util/cdkHelpers.js:199]: Let's rename `nonPluginsKey` to something more descriptive ie. `unallowedKey`

Comment (KinnersleyM) [reply packages/cli/scripts/util/cdkHelpers.js]: Sorted

**Comment by fwang** [comment]
Thanks @KinnersleyM, will take a look at it today!

**Comment by fwang** [comment]
@KinnersleyM with v0.34.7, you can now configure `keepNames` inline like this:
```
  bundle: {
    esbuildConfig: {
      keepNames: true,
    },
  },
```

Let me know if that makes sense.

**Pull request merged**